open-metadata-implementation/adapters/open-connectors/configuration-store-connectors/configuration-file-store-connector/src/main/java/org/odpi/openmetadata/adapters/adminservices/configurationstore/file/FileBasedUIServerConfigStoreConnector.java

```diff
@@ -0,0 +1,146 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+package org.odpi.openmetadata.adapters.adminservices.configurationstore.file;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.odpi.openmetadata.frameworks.connectors.properties.ConnectionProperties;
+import org.odpi.openmetadata.frameworks.connectors.properties.EndpointProperties;
+import org.odpi.openmetadata.userinterface.adminservices.configuration.properties.UIServerConfig;
+import org.odpi.openmetadata.userinterface.adminservices.store.UIServerConfigStoreConnectorBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+
+public class FileBasedUIServerConfigStoreConnector extends UIServerConfigStoreConnectorBase
+{
+    /*
+     * This is the name of the configuration file that is used if there is no file name in the connection.
+     */
+    private static final String defaultFilename = "ui.server.config";
+
+    /*
+     * Variables used in writing to the file.
+     */
+    private String           configStoreName  = null;
+
+    /*
+     * Variables used for logging and debug.
+     */
+    private static final Logger log = LoggerFactory.getLogger(FileBasedUIServerConfigStoreConnector.class);
+
+
+    /**
+     * Default constructor
+     */
+    public FileBasedUIServerConfigStoreConnector()
+    {
+    }
+
+
+    @Override
+    public void initialize(String connectorInstanceId, ConnectionProperties connectionProperties)
+    {
+        super.initialize(connectorInstanceId, connectionProperties);
+
+        EndpointProperties endpoint = connectionProperties.getEndpoint();
+
+        if (endpoint != null)
+        {
+            configStoreName = endpoint.getAddress();
+        }
+
+        if (configStoreName == null)
+        {
+            configStoreName = defaultFilename;
+        }
+    }
+
+
+    /**
+     * Save the server configuration.
+     *
+     * @param omagServerConfig - configuration properties to save
+     */
+    public void saveServerConfig(UIServerConfig omagServerConfig)
```

Comment:
OMAG stands for Open Metadata and Governance so it is ok to use for naming in UI - however need to be consistent as to whether it is any server on either platform or the server running on the server platform

Reply:
corrected the variable name for this checkin